include/npcomp/Dialect/Basicpy/IR/BasicpyOps.td

```diff
@@ -90,7 +90,65 @@ def CompareOperationAttr : StrEnumAttr<
 }
 
 //===----------------------------------------------------------------------===//
-// Ops
+// Ops that parallel Python bytecodes
+// Where possible and semantics allow, prefer to structure ops for compatibility
+// and similar naming as those used in the CPython VM.
+//===----------------------------------------------------------------------===//
+
+// TODO: Implement ConstantLike op trait.
+def Basicpy_BuildDictOp : Basicpy_Op<"build_dict", [NoSideEffect]> {
+  let summary = "Builds an empty dict";
+  let description = [{
+    This op mirrors the CPython BUILD_MAP op (note naming difference).
+
+    Note that as with CPython, this op only builds an empty dict; however,
+    it is reserved in the future for it to take variadic operands to construct
+    with a list of key/value pairs.
+  }];
+  let arguments = (ins
+  );
+  let results = (outs
+    Basicpy_ListType:$result
+  );
+  let assemblyFormat = "operands attr-dict `:` functional-type(operands, results)";
+}
+
+// TODO: Implement ConstantLike op trait.
+def Basicpy_BuildListOp : Basicpy_Op<"build_list", [NoSideEffect]> {
```

Comment:
General comment: would be nice to see the new types/ops introduced in this commit to at least show up at least in a basic "ops.mlir" kind of test on the mlir side.